beacon_node/execution_layer/src/lib.rs

```diff
@@ -535,64 +568,121 @@ impl ExecutionLayer {
     ///
     /// The result will be returned from the first node that returns successfully. No more nodes
     /// will be contacted.
-    pub async fn get_payload<T: EthSpec>(
+    pub async fn get_payload<T: EthSpec, Txns: Transactions<T>>(
         &self,
         parent_hash: ExecutionBlockHash,
         timestamp: u64,
         prev_randao: Hash256,
         finalized_block_hash: ExecutionBlockHash,
         proposer_index: u64,
-    ) -> Result<ExecutionPayload<T>, Error> {
+    ) -> Result<ExecutionPayload<T, Txns>, Error> {
         let _timer = metrics::start_timer_vec(
             &metrics::EXECUTION_LAYER_REQUEST_TIMES,
             &[metrics::GET_PAYLOAD],
         );
 
         let suggested_fee_recipient = self.get_suggested_fee_recipient(proposer_index).await;
 
-        debug!(
-            self.log(),
-            "Issuing engine_getPayload";
-            "suggested_fee_recipient" => ?suggested_fee_recipient,
-            "prev_randao" => ?prev_randao,
-            "timestamp" => timestamp,
-            "parent_hash" => ?parent_hash,
-        );
-        self.engines()
-            .first_success(|engine| async move {
-                let payload_id = if let Some(id) = engine
-                    .get_payload_id(parent_hash, timestamp, prev_randao, suggested_fee_recipient)
+        match Txns::block_type() {
+            BlockType::Blinded => {
+                debug!(
+                    self.log(),
+                    "Issuing engine_getPayloadHeader";
+                    "suggested_fee_recipient" => ?suggested_fee_recipient,
+                    "prev_randao" => ?prev_randao,
+                    "timestamp" => timestamp,
+                    "parent_hash" => ?parent_hash,
+                );
```

Comment:
Whoops there's a couple mistakes here.. I accidentally deleted the full block log here, probably during a merge with unstable. And the two logs are slightly different because the endpoints they hit have different names. This one should be `builder_getPayloadHeader` and the other one should be `engine_getPayload`